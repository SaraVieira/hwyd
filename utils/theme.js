import {
  /* colors */
  APPLE_VALLEY,
  BREWED_MUSTARD,
  CREAMY_PEACH,
  OLD_GERANIUM,
  PENCIL_LEAD,
  ROGUE_PINK,
  ROSY_HIGHLIGHT,
  SOFT_BLUE,
  SQUEAKY,
  WHITE,

  /* fonts */
  BOLD_FONT,
  ITALIC_FONT,
  REGULAR_FONT,
  SUBTITLE_FONT,
  TITLE_FONT,
} from './constants';

export default {
  colors: {
    appleValley: APPLE_VALLEY,
    brewedMustard: BREWED_MUSTARD,
    creamyPeach: CREAMY_PEACH,
    oldGeranium: OLD_GERANIUM,
    pencilLead: PENCIL_LEAD,
    roguePink: ROGUE_PINK,
    rosyHighlight: ROSY_HIGHLIGHT,
    softBlue: SOFT_BLUE,
    squeaky: SQUEAKY,
    white: WHITE,
  },
  fonts: {
    bold: BOLD_FONT,
    italic: ITALIC_FONT,
    regular: REGULAR_FONT,
    subtitle: SUBTITLE_FONT,
    title: TITLE_FONT,
  },
};
