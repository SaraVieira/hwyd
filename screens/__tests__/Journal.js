import React from 'react';
import { ThemeProvider } from 'styled-components';
import { shallow } from 'enzyme';
import 'jest-styled-components';

import Journal from '../Journal';
import theme from '../../utils/theme';

describe('<Journal />', () => {
  it('renders with no notes', () => {
    const tree = shallow(
      <ThemeProvider theme={theme}>
        <Journal />
      </ThemeProvider>,
    );

    const root = tree.find(Journal).shallow();
    expect(root).toMatchSnapshot();
  });

  it('renders a single note', () => {
    const tree = shallow(
      <ThemeProvider theme={theme}>
        <Journal />
      </ThemeProvider>,
    );

    const root = tree.find(Journal).shallow();
    root.setState({
      notes: [
        { date: '03-02-2018', note: 'Today was...', result: 'GOOD', id: 0 },
      ],
    });
    expect(root).toMatchSnapshot();
  });

  it('renders a carousel of notes', () => {
    const tree = shallow(
      <ThemeProvider theme={theme}>
        <Journal />
      </ThemeProvider>,
    );

    const root = tree.find(Journal).shallow();
    root.setState({
      notes: [
        { date: '03-02-2018', note: 'Today was...', result: 'GOOD', id: 0 },
        { date: '03-03-2018', note: 'Today was...', result: 'GOOD', id: 1 },
        { date: '03-04-2018', note: 'Today was...', result: 'BAD', id: 2 },
        { date: '03-05-2018', note: 'Today was...', result: 'BAD', id: 3 },
      ],
    });
    expect(root).toMatchSnapshot();
  });
});
