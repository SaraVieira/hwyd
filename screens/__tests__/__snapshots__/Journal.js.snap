// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Journal /> renders a carousel of notes 1`] = `
<Styled(View)>
  <BackgroundGradient />
  <Styled(Text)>
    Journal
  </Styled(Text)>
  <Styled(SideSwipe)
    contentContainerStyle={
      Object {
        "alignItems": "center",
      }
    }
    contentOffset={209}
    data={
      Array [
        Object {
          "date": "03-02-2018",
          "id": 0,
          "note": "Today was...",
          "result": "GOOD",
        },
        Object {
          "date": "03-03-2018",
          "id": 1,
          "note": "Today was...",
          "result": "GOOD",
        },
        Object {
          "date": "03-04-2018",
          "id": 2,
          "note": "Today was...",
          "result": "BAD",
        },
        Object {
          "date": "03-05-2018",
          "id": 3,
          "note": "Today was...",
          "result": "BAD",
        },
      ]
    }
    itemWidth={332}
    renderItem={[Function]}
    threshold={100}
  />
</Styled(View)>
`;

exports[`<Journal /> renders a single note 1`] = `
<Styled(View)>
  <BackgroundGradient />
  <Styled(Text)>
    Journal
  </Styled(Text)>
  <Styled(View)>
    <JournalEntry
      date="03-02-2018"
      id={0}
      note="Today was..."
      result="GOOD"
    />
  </Styled(View)>
</Styled(View)>
`;

exports[`<Journal /> renders with no notes 1`] = `
<Styled(View)>
  <BackgroundGradient />
  <Styled(Text)>
    Journal
  </Styled(Text)>
  <Styled(Text)>
    No entries yet.
  </Styled(Text)>
</Styled(View)>
`;
